pkgs/tools/games/alice-tools/default.nix

```diff
@@ -0,0 +1,108 @@
+{ stdenv
+, lib
+, gitUpdater
+, fetchFromGitHub
+, fetchpatch
+, meson
+, ninja
+, pkg-config
+, bison
+, flex
+, libiconv
+, libpng
+, libjpeg
+, libwebp
+, zlib
+, withGUI ? true
+, qtbase
+, qttools
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "alice-tools";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "nunuhara";
+    repo = "alice-tools";
+    rev = version;
+    fetchSubmodules = true;
+    hash = "sha256-1r+TUGY2eSuHzz7G+YDhOgZgOmCIRhFSIfIdsYFsc0k=";
+  };
+
+  patches = [
+    # Fixes macOS Libsystem headers confused by 'noreturn' define in system4 & alice-tools
+    # https://github.com/nunuhara/alice-tools/issues/45
+    ./0001-Fix-noreturn-define-confusion-on-macOS.patch
+
+    # Fixes compilation with format hardening
+    # Remove when nunuhara/alice-tools/pull/43 merged & in a release
+    (fetchpatch {
+      url = "https://github.com/nunuhara/alice-tools/pull/43/commits/95748c7f825b70b998c04d73ace31f8fe6150adc.patch";
+      hash = "sha256-lHxUgSFifScnoNBSNedc9RjyF+EWuGYFM4xKtV5SHgk=";
+    })
+
+    # Fixes incorrect template class initialisation aborting compilation on Darwin
+    # Remove when nunuhara/alice-tools/pull/44 merged & in a release
+    (fetchpatch {
+      url = "https://github.com/nunuhara/alice-tools/pull/44/commits/ad962a53a0a952122d107b8f710818088777d4f1.patch";
+      hash = "sha256-euS/bhCmfOR/vEBCFHQBicMNDfjiw+9cvr4bf+2qHPk=";
+    })
+  ];
+
+  postPatch = lib.optionalString (withGUI && lib.versionAtLeast qtbase.version "6.0") ''
+    substituteInPlace src/meson.build \
+      --replace qt5 qt6
+
+    # Don't know if this is a bug in Qt6 or meson but moc, uic & rrc don't get found without this
+    export PATH=${qtbase.dev}/libexec:$PATH
```

Comment:
With Qt5 detection works just fine:
```
Run-time dependency qt5 (modules: Core, Gui, Widgets) found: YES 5.15.7 (pkg-config)
Detecting Qt5 tools
Run-time dependency qt5 (modules: Core) found: YES 5.15.7 (pkg-config)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/moc found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/moc)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/uic found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/uic)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/rcc found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/rcc)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/lrelease found: NO
Program lrelease5 found: NO
Program lrelease-qt5 found: NO
Program lrelease found: YES 5.15.7 (/nix/store/fs86kagqgwn6iabm13hxh0fifxgyfw6c-qttools-5.15.7-dev/bin/lrelease)
```

With Qt6, it doesn't:
```
Run-time dependency qt6 (modules: Core, Gui, Widgets) found: YES 6.4.1 (pkg-config)
Detecting Qt6 tools
Run-time dependency qt6 (modules: Core) found: YES 6.4.1 (pkg-config)
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/moc found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/moc found: NO
Program moc6 found: NO
Program moc-qt6 found: NO
Program moc found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/uic found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/uic found: NO
Program uic6 found: NO
Program uic-qt6 found: NO
Program uic found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/rcc found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/rcc found: NO
Program rcc6 found: NO
Program rcc-qt6 found: NO
Program rcc found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/lrelease found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/lrelease found: NO
Program lrelease6 found: NO
Program lrelease-qt6 found: NO
Program lrelease found: YES 6.4.1 (/nix/store/0bjv6cn2qr4lg9j13gxydi7n65qipyzf-qttools-6.4.1-dev/bin/lrelease)

src/meson.build:129:4: ERROR: MOC sources specified and couldn't find uic-qt6, please check your qt6 installation
```

In our Qt6 package `moc`, `uic` and `rrc` are in `${qtbase.dev}/libexec` while in Qt5 they were in `${qtbase.dev}/bin`. Meson looks in `${qtbase.dev}/bin`, `${qtbase}/libexec` and `$PATH` before giving up. Not sure if the bins are supposed to be in `${qtbase.dev}/bin` or the Meson Qt6 detection needs to look in  `${qtbase.dev}/libexec` instead.

Reply:
I've opened #210134 to fix this. Once this PR is merged, I'll open another PR to undo the hack in staging

Reply:
(Building this PR with the hack removed but on top of that other PR does in fact build -- so yeah this seems to be the fix)

Reply:
The fix for this just hit master. Can you see if it works against master without this hack now?

Reply:
Yep works fine now, will update.

Reply:
Sweet, if ofborg builds then this PR looks fabulous to me so I'll approve and then see if we can get a committer to review/merge it